### PR TITLE
Add map, filter, some, every, and groupBy

### PR DESCRIPTION
**Pull request by robinpokorny**
These are some methods I always wished were available for Sets 😊

### CODE REVIEW

File: metho-set.js [hunk 1]
```diff
@@ -27,3 +27,54 @@ export const difference = Metho.add(
   }
 )
 
+// Map items
+export const map = Metho.add(
+  target,
+  function map(fn) {
+    return new Set([...this].map(i => fn(i)));
```
Comment (jonrandy) [inline metho-set.js]: Also pass the set itself into the callback function - to mirror the additional parameters that array method callbacks receive (obviously passing in `index` makes no sense for sets). This should apply to all of these new methods.

We can also drop the semi-colons... I'm not sure if I've been consistent with this in the Metho code, but I prefer to not have them

**Review by jonrandy — CHANGES_REQUESTED** [review]
Thanks for this - these are a good idea. I've left some comments for a few minor changes.

File: metho-set.js (hunk 1 above)
Comment (robinpokorny) [reply metho-set.js]: Good point! Should I pass the value twice and then the set to be consistent with [Set#forEach](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set/forEach)?

Ah, the semicolons are an oversight. Will remove them!

Comment (jonrandy) [reply metho-set.js]: Yeah, that's a good idea for consistency 👍

**Comment by robinpokorny** [comment]
Updated the PR, please have a look!

**Comment by jonrandy** [comment]
LGTM - will merge and update the version etc. on NPM at the weekend. Thanks!

**Pull request merged**